src/persist/src/storage.rs

```diff
@@ -21,25 +21,30 @@ use futures_executor::block_on;
 use crate::error::Error;
 use crate::indexed::encoding::BlobMeta;
 
-/// Sanity check whether we can decode the Blob's persisted meta object.
+/// Sanity check whether we can decode the Blob's persisted meta object, and delete
+/// all data if the encoded version is less than what the current implementation supports.
 ///
 /// TODO: this is a hack and we will need to get rid of this once we have a
 /// proper backwards compatibility policy.
-pub fn check_meta_version<B: Blob>(b: &mut B) -> Result<(), Error> {
+pub fn check_meta_version_maybe_delete_data<B: Blob>(b: &mut B) -> Result<(), Error> {
     let meta = match block_on(b.get("META"))? {
         None => return Ok(()),
         Some(bytes) => bytes,
     };
 
     let current_version = BlobMeta::version();
-    let persisted_version = BlobMeta::encoded_version(&meta)
-        .ok_or_else(|| Error::from("unable to determine previously persisted encoding version."))?;
+    let persisted_version = BlobMeta::encoded_version(&meta)?;
 
     if current_version == persisted_version {
         // Nothing to do here, everything is working as expected.
         Ok(())
     } else if current_version > persisted_version {
         // Delete all the keys, as we are upgrading to a new version.
+        log::info!(
+            "Deleting all previously persisted data as part of upgrade from version {} to {}.",
```

Comment:
just to stay ahead of it, this needs to also somehow explain to the user that this will only happen while persistence is in beta

Reply:
I added the text "Persistence beta detected a version mismatch. " to the start of this log output which i think accomplishes that.